plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReLoader.java

```diff
@@ -0,0 +1,273 @@
+        CompletableFuture.runAsync(() -> autoReLoadModelByNodeId(localNodeId));
+    }
+
+    /**
+     * auto reload all the models under the node id
+     * the node must be a ml node
+     * @param nodeId node id
+     */
+    @VisibleForTesting
+    void autoReLoadModelByNodeId(String nodeId) {
+        if (!isExistedIndex(ML_TASK_INDEX)) {
+            return;
+        }
+
+        SearchResponse response = queryTask(nodeId);
+
+        SearchHit[] hits = response.getHits().getHits();
+        if (CollectionUtils.isEmpty(hits)) {
+            return;
+        }
+
+        if (hits.length != 1) {
+            log.error("Can't get the latest successful deployment in node id {}", nodeId);
+        }
+
+        int reTryTimes = 0;
+        try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, hits[0].getSourceRef())) {
+            ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+            MLTask mlTask = MLTask.parse(parser);
+
+            autoReLoadModelByNodeAndModelId(nodeId, mlTask.getModelId());
+        } catch (RuntimeException | IOException e) {
+            reTryTimes++;
+            // Store the latest value of the reTryTimes and node id under the index ".plugins-ml-model-reload"
+            saveLatestReTryTimes(nodeId, reTryTimes);
+            log.error("Can't auto reload model in node id {} ,has try {} times\nThe reason is:{}", nodeId, reTryTimes, e.getMessage());
+        }
+
+        // Store the latest value of the reTryTimes and node id under the index ".plugins-ml-model-reload"
+        saveLatestReTryTimes(nodeId, reTryTimes);
+    }
+
+    /**
+     * query task index, and get the result of "task_type"="LOAD_MODEL" and "state"="COMPLETED"
+     * and "worker_node" match nodeId
+     * @param nodeId one of query condition
+     * @return SearchResponse
+     */
+    @VisibleForTesting
+    SearchResponse queryTask(String nodeId) {
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().from(0).size(1);
+
+        QueryBuilder queryBuilder = QueryBuilders
+            .boolQuery()
+            .must(QueryBuilders.matchPhraseQuery("task_type", "LOAD_MODEL"))
+            .must(QueryBuilders.matchPhraseQuery("state", "COMPLETED"))
+            .must(QueryBuilders.matchPhraseQuery("worker_node", nodeId));
+        searchSourceBuilder.query(queryBuilder);
+
+        SortBuilder sortBuilderOrder = new FieldSortBuilder("last_update_time").order(SortOrder.DESC);
+        searchSourceBuilder.sort(sortBuilderOrder);
+
+        SearchRequest searchRequest = new SearchRequest(ML_TASK_INDEX);
+        searchRequest.source(searchSourceBuilder).indices(ML_TASK_INDEX);
+        try {
+            return client.execute(SearchAction.INSTANCE, searchRequest).actionGet(5000);
+        } catch (IndexNotFoundException e) {
+            throw new IndexNotFoundException("index " + ML_TASK_INDEX + " not found");
```

Comment:
Add logging here.

Reply:
done

Reply:
When logging exception, log the exception itself like `log.error("xxxxxx", e)`

Reply:
done